upload-cloud-storage/src/client.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import { UploadHelper } from './upload-helper';
+import { Storage, UploadResponse } from '@google-cloud/storage';
+
+/**
+ * Available options to create the client.
+ *
+ * @param credentials GCP JSON credentials (default uses ADC).
+ */
+
+type ClientOptions = {
+  credentials?: string;
+};
+
+/**
+ * Handles credential lookup, registration and
+ * wraps interactions with the GCS Helper
+ *
+ * @param opts list of ClientOptions
+ * @returns Client
+ */
+export class Client {
+  readonly storage: Storage;
+  constructor(opts?: ClientOptions) {
+    if (opts?.credentials) {
+      // If the credentials are not JSON, they are probably base64-encoded. Even
+      // though we don't instruct users to provide base64-encoded credentials,
+      // sometimes they still do.
+      if (!opts.credentials.trim().startsWith('{')) {
+        const creds = opts.credentials;
+        opts.credentials = Buffer.from(creds, 'base64').toString('utf8');
+      }
+      const creds = JSON.parse(opts.credentials);
+      this.storage = new Storage({ credentials: creds });
```

Comment:
We'll need to figure out a way to plump in a user agent string here.

Reply:
We may need to follow up with the [nodejs-storage](https://github.com/googleapis/nodejs-storage) team. I dont see any options jumping out in the docs :(

Reply:
Understood. Let's put in a TODO here. Perhaps open up an issue with the project making the request.